Significantly increases stamina recovery on most effects that provide it

Metekillot opened the pull request:
## About The Pull Request

Currently, stamina healing is a niche side effect on many drugs, besides a few effects (mostly heretic powers, an obscure side effect of some drinks) that actually contribute to a significant shift in gameplay style when you use them. In this PR I significantly increased the stamina recovery on almost every drug and some effects that provide it. 

Points of interest (not completely inclusive):
Syndicate stimulants heal 12 instead of 5 stamina damage. This will bring you out of stam crit in two cycles, and let you heal out of the slowdown from the stamina damage from a baton strike. The current rate of 5 takes 5 seconds to heal out of stam crit, and the stamina reset takes place before you heal out of stamina damage slowdown.
Bath salts heal stamina at the rate stimulants used to. Meth heals stamina slightly slower.
Changeling adrenaline will instantly bring you out of stam crit if you're in it; otherwise, it will heal 10 stamina damage.
Ephedrine heals 4 stamina per life, instead of 1.
Being so drunk that you black out and give up your body to ghost control will make you heal stamina more quickly than an ascended rust heretic. You will also likely be dying of liver damage.
## Why It's Good For The Game

At the levels they are set, most of these stamina healing effects have almost no effect on gameplay whatsoever except in the case of tanking a couple of extra disabler shots (only in the case of the high end drugs), or being able to do stamina consuming tasks slightly longer (breaking boulders, grinding with pestle and mortar, working out, ???). 

Adjusting these numbers would make these drugs an actual factor in combat situations that rely on stamina crit (stun batons, ebows, stamina damaging poisons and diseases). Most of the drugs already have side effects such as brain and organ damage, which in the current state hugely outweighs any reason to use most of them besides a few that give a movespeed buff or prevent baton knockdown (ephedrine, stimulants).

This only affects stamina healing. There has been no change to any other interactions with things that knock down or cause any other kind of stun.

These changes are in consideration of the fact that taking stamina damage puts you in a cooldown before you can recover your stamina naturally. Effects that cause steady stamina damage will bottom out your stamina the way they do now, unless you use something to heal it.
## Changelog
:cl:
balance: Stamina healing has been significantly increased for most reagents and passive effects.
/:cl:

Metekillot committed:
significantly increases stamina recovery on most effects that provide it

Metekillot committed:
nudged the values to make sure that it takes a certain number of seconds for some drugs to heal you out of stam crit

Metekillot committed:
significantly lowered turbo heal rate because i made it the strongest stam regen in the game on accident :)

LemonInTheDark commented:
We have to some extent been here before <https://github.com/tgstation/tgstation/pull/46211>

MrMelbert commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1351,7 +1351,9 @@
 	. = ..()
 	metabolizer.AdjustAllImmobility(-20 * REM * seconds_per_tick)
-	if(metabolizer.adjustStaminaLoss(-10 * REM * seconds_per_tick, updating_stamina = FALSE))
+	// If stamina critted, brings them out of it; otherwise, heals (quite a bit) of stamina
+	// damage without totally negating the slowness of having taken stamina damage
+	if(metabolizer.adjustStaminaLoss(metabolizer.getStaminaLoss() > 100 ? -21 : -10 * REM * seconds_per_tick, updating_stamina = FALSE))
```
This is just goofy, you might as well just make them immune to stamcrit at this point

Metekillot replied:
This is to make them vulnerable to stamina slowdown. You slow if you have more than 40 stam damage. Having it a static 20 or 21 seemed too much, but the situation of being stunlocked by thugs with baton was the main killer of lings. If they're an instant round removal threat, they should be more threatening, without being immune to stamina slowdown.

MrMelbert replied:
I didn't say make them immune to stamina slowdown, I said make them immune to stamcrit

MrMelbert commented on code/datums/status_effects/debuffs/spacer.dm:
```diff
@@ -40,9 +40,9 @@
 	if(seconds_in_nograv >= 3 MINUTES)
 		// This has some interesting side effects with gravitum or similar negating effects that may be worth nothing
-		owner.adjustStaminaLoss(-1 * stamina_heal_per_tick)
+		owner.adjustStaminaLoss(-3 * stamina_heal_per_tick)
```
1. Why are you changing the the multiplier and not the var
2. I don't think spacers should be getting passive 9 stamina regen per tick, that seems too much

Metekillot replied:
1. Because I can't read good and I wanna learn to do other stuff good
2. Spacer isn't good enough to cost 7 trait points but, I agree. This was meant to make it 2 higher per tick, not 9...

Metekillot replied:
actually just leaving it how it was; 3 stam per tick seems reasonable for something that you could have perpetually if you blow up the grav gen or do the unthinkable of actually getting wittel

Formerxboxgamer commented:
Adrenal buff is pretty scary, stamcrit is one of the few ways to reliably kill chitin murderbone lings.

Deadgebert commented:
That's a pretty crazy buff to stims, they'll have to be re-stun batoned within 4 seconds of last hit. cuffing takes 3 seconds with black gloves. can i suggest 9 instead of 12 for 1 extra tick of leeway? other than that, looks good to me :>

paganiy commented:
> Adrenal buff is pretty scary, stamcrit is one of the few ways to reliably kill chitin murderbone lings.

Isn't it only natural that you can't stop an immortal scary creature on stimulants with 2 hit winstick? Use lethals. Ling's regeneration is not strong enough to withstand 16 laser shots of 25 damage each. If they used EMP, then use the shotgun until their chemicals run out. Shrimple.

Admiral-Gigaton commented:
"Man too drunk to die"

> Adrenal buff is pretty scary, stamcrit is one of the few ways to reliably kill chitin murderbone lings.

We were born to inherit the stars.

YuliceTheDruid commented:
> If they used EMP, then use the shotgun until their chemicals run out. Shrimple.

This man does not know.

projectkepler-ru commented:
>  then use the shotgun until their chemicals run out. Shrimple.

with what ammo

Metekillot commented:
> Adrenal buff is pretty scary, stamcrit is one of the few ways to reliably kill chitin murderbone lings.

Being able to reliably kill lings with winstick shuttlegib true combo seems lame if they're a threat where the SOP is to RR them the moment you're qble

Metekillot commented:
> That's a pretty crazy buff to stims, they'll have to be re-stun batoned within 4 seconds of last hit. cuffing takes 3 seconds with black gloves. can i suggest 9 instead of 12 for 1 extra tick of leeway? other than that, looks good to me :>

My intent was for chem purging or lethals to be the way to subdue stim users, since they're 1/4 of a round start tater tot's TC, but it that's truly too good then adjusting the rate isn't a problem

Metekillot commented:
Please remember turbo stam Regen doesn't affect vomit stuns or flashbang stuns or laser crits :')

Deadgebert commented:
> > That's a pretty crazy buff to stims, they'll have to be re-stun batoned within 4 seconds of last hit. cuffing takes 3 seconds with black gloves. can i suggest 9 instead of 12 for 1 extra tick of leeway? other than that, looks good to me :>
> 
> My intent was for chem purging or lethals to be the way to subdue stim users, since they're 1/4 of a round start tater tot's TC, but it that's too strong then adjusting the rate isn't a problem

i was thinking more about nukies. nukies smart enough to buy 12tc medkit are really hard to kill with straight damage. like extremely difficult.
rethinking this, the required time between stun baton hits is actually between 2 and 4 seconds, changing it to 9 would be between 4 and 6 seconds, assuming they don't have anything else that gives stam regen.

I agree that 1/4th traitor TC is a lot for 5 minutes, but stim pen is pretty strong as is. maybe the better solution would be reducing the cost? It already takes 3 stun baton hits to stam crit and they ignore the stun baton knockdown.

Metekillot commented:
If you're referring to the current stats, 2 stun baton hits stamcrit a stims user unless it has been 8-9 seconds (they need to recover 25 stamina to not get stamcrit by a second baton hit). Stamina resets to full happens at 10 seconds.

12 would make that 3-4 seconds of avoidance of a subsequent baton hit, and 9 would make that 5-6.

xzero314 commented:
This combined with the PR to give syndie mods a free baton tank is going to be pretty game breaking.  Have you tried stims lately? They allow you to take a baton hit and not get knocked down.  I cant imagine how hard its going to be to kill a traitor/nukie rocking a syndie mod with free baton tank, stims which are also a free baton tank, block chance via esword/eshield or d sword, CNS rebooter to undo  getting by all of that once per minute. Throw on the shield module if they are a nukie. How are you supposed to fight that short of bombs?

Metekillot commented:
> This combined with the PR to give syndie mods a free baton tank is going to be pretty game breaking.  Have you tried stims lately? They allow you to take a baton hit and not get knocked down.  I cant imagine how hard its going to be to kill a traitor/nukie rocking a syndie mod with free baton tank, stims which are also a free baton tank, block chance via esword/eshield or d sword, CNS rebooter to undo  getting by all of that once per minute. Throw on the shield module if they are a nukie. How are you supposed to fight that short of bombs?

If you're asking in earnest:

Bombs
Purge chems and poisons
Non energy ranged weapons
Flashbangs to knockdown and disarm
Acid
Traps 
Mechs 
Mobbing
EMPs

For your specific case of "spent every single TC on this specific kit"

Assuming that the syndie mod baton resist builtin module PR gets merged. I think it is a bit of a conclusion to jump to to presume that another PR is going to get merged as an argument inside another unmerged PR.

carlarctg commented:
stun baton nerf, close

Metekillot committed:
spacer recovery reversion, ling adrenal rework

reverted spacer stam restore rate to what it was before. changeling adrenaline now makes you immune to stamcrit, but if you do hit stamcrit with it in you, it burns off 1 unit (or the rest of it if you have less than 1 unit left). added logic to stamcrit itself for signal listeners that might want to cancel stamcrit.

Metekillot commented:
> This combined with the PR to give syndie mods a free baton tank is going to be pretty game breaking. Have you tried stims lately? They allow you to take a baton hit and not get knocked down. I cant imagine how hard its going to be to kill a traitor/nukie rocking a syndie mod with free baton tank, stims which are also a free baton tank, block chance via esword/eshield or d sword, CNS rebooter to undo getting by all of that once per minute. Throw on the shield module if they are a nukie. How are you supposed to fight that short of bombs?

By the way, to readress what you said about "free baton tank". Baton resistance makes you resist KNOCKDOWN. You still take stamina damage from it. You still go into stamcrit if you drop below 0 stamina (unless you have ling adrenaline, and in that case, it burns off a unit of your adrenaline in exchange for 10 more stamina to keep you on your feet).

Metekillot committed:
moved a define into combat.dm

Metekillot committed:
ling adrenaline sets your stamina to 10 if you hit stamcrit, not recovers 10 stamina

Metekillot committed:
Merge branch 'master' into stamina-heal-buff

Formerxboxgamer commented:
> > This combined with the PR to give syndie mods a free baton tank is going to be pretty game breaking.  Have you tried stims lately? They allow you to take a baton hit and not get knocked down.  I cant imagine how hard its going to be to kill a traitor/nukie rocking a syndie mod with free baton tank, stims which are also a free baton tank, block chance via esword/eshield or d sword, CNS rebooter to undo  getting by all of that once per minute. Throw on the shield module if they are a nukie. How are you supposed to fight that short of bombs?
> 
> If you're asking in earnest:
> 
> Bombs Purge chems and poisons Non energy ranged weapons Flashbangs to knockdown and disarm Acid Traps Mechs Mobbing EMPs
> 
> For your specific case of "spent every single TC on this specific kit"
> 
> Assuming that the syndie mod baton resist builtin module PR gets merged. I think it is a bit of a conclusion to jump to to presume that another PR is going to get merged as an argument inside another unmerged PR.

Half of these don’t work. Nukies have EMP shielding. You’d need piercing syringes to purge, so lol and lmao you’re not getting that on war. Non energy based ranged weapons? So like pipeguns? We removed lethal shotgun and you wouldn’t ever get close to stamcritting with slugs if they had this kind of regen. Flashbangs? Against nukie hardsuits? You’re trolling

Metekillot commented:
> > > This combined with the PR to give syndie mods a free baton tank is going to be pretty game breaking.  Have you tried stims lately? They allow you to take a baton hit and not get knocked down.  I cant imagine how hard its going to be to kill a traitor/nukie rocking a syndie mod with free baton tank, stims which are also a free baton tank, block chance via esword/eshield or d sword, CNS rebooter to undo  getting by all of that once per minute. Throw on the shield module if they are a nukie. How are you supposed to fight that short of bombs?
> > 
> > If you're asking in earnest:
> > 
> > Bombs Purge chems and poisons Non energy ranged weapons Flashbangs to knockdown and disarm Acid Traps Mechs Mobbing EMPs
> > 
> > For your specific case of "spent every single TC on this specific kit"
> > 
> > Assuming that the syndie mod baton resist builtin module PR gets merged. I think it is a bit of a conclusion to jump to to presume that another PR is going to get merged as an argument inside another unmerged PR.
> 
> Half of these don’t work. Nukies have EMP shielding. 

Good point. I need to see if that's built in or a module they have to buy.

> You’d need piercing syringes to purge, so lol and lmao you’re not getting that on war.

Why not? Piercing syringes are easily doable in 25 minutes.

> Non energy based ranged weapons? So like pipeguns? We removed lethal shotgun and you wouldn’t ever get close to stamcritting with slugs if they had this kind of regen. 

Tech shells, weapons development tech, and other ammo are available 11 minutes into the shift if science is strictly ran by an RD or Captain, which it likely would be in this war scenario. They're available even earlier if experiments are performed. 11 minutes is without experiments, at a cost of 35000 research points.

>Flashbangs? Against nukie hardsuits You’re trolling

Flashbangs are a special snowflake sec item. They knock down through any protection at a minimum if detonated at close range.

> 
> Edit: fairly certain acid is garbage against nukie suits too. 

You may have me there. I know it has SOME acid protection; I'll have to verify myself whether it's complete.

> How about we just ask the fast moving knockdown immune machine gun wielding guy with 100 tc worth of gear to kite correctly and watch their positioning instead of buffing them so that it’s brain dead easy to win?

They're not knockdown immune as has been stated both in this comment, several others (by me), and the code. You're correct that you usually need to git gud to win as nukie.

Loafin34 commented:
based.... stimulants are supposed to be the answer to batongs/disabler fire, but i do worry about ye old dsword meth-stimmed no slip man getting even more power

Deadgebert commented:
Yes, there technically are *other* solutions besides stun baton, but that doesn't mean they're reliable enough to use. Stunning is the primary way of dealing with a nukie. It's really the only threat outside of bombs and getting crushed by a firelock. Your other suggested solutions require near perfect cooperation from other parties or are completely negated by easily obtainable items. 

Purging chems are not really a valid solution to stam recovery drugs, it's too hard to get it in the other person's system, they have too many ways to counteract the delivery methods outside of piercing syringe. Even piercing syringe aren't super great, in every map except north star and delta there is only one syringe gun outside of illegal weapon research. You have one person with the ability to shoot them with purge chems.

Flashbangs and lube are good ways to disarm or steal weapons, sure, but is that really the only way we should be able to deal with a threat?

Stamina and stam crit are currently a core part of the game. I hate to say it, but stun combat is not in a bad state. Game has largely been balanced around crew having stun weaponry and not much else, taking it away without providing an alternative is just going to lead to neigh unstoppable antags. 

This PR just feels like an ided. changelings probably need a buff and stim tc cost should probably be reduced, but i think that's about it.

SimplySmiley123 commented:
> > Adrenal buff is pretty scary, stamcrit is one of the few ways to reliably kill chitin murderbone lings.
> 
> Isn't it only natural that you can't stop an immortal scary creature on stimulants with 2 hit winstick? Use lethals. Ling's regeneration is not strong enough to withstand 16 laser shots of 25 damage each. If they used EMP, then use the shotgun until their chemicals run out. Shrimple.

The shotguns are a meme for crew tho.

Loafin34 commented:
bring back bootknives...

SimplySmiley123 commented:
> > This combined with the PR to give syndie mods a free baton tank is going to be pretty game breaking.  Have you tried stims lately? They allow you to take a baton hit and not get knocked down.  I cant imagine how hard its going to be to kill a traitor/nukie rocking a syndie mod with free baton tank, stims which are also a free baton tank, block chance via esword/eshield or d sword, CNS rebooter to undo  getting by all of that once per minute. Throw on the shield module if they are a nukie. How are you supposed to fight that short of bombs?
> 
> If you're asking in earnest:
> 
> Bombs Purge chems and poisons Non energy ranged weapons Flashbangs to knockdown and disarm Acid Traps Mechs Mobbing EMPs
> 
> For your specific case of "spent every single TC on this specific kit"
> 
> Assuming that the syndie mod baton resist builtin module PR gets merged. I think it is a bit of a conclusion to jump to to presume that another PR is going to get merged as an argument inside another unmerged PR.

Non-energy based weapons ain't something crew can get now without a traitor's emag. Shotguns atm are non-lethal, the lethal shells crew can feasibly get, are things nukies get to ignore entirely (fire and dart), with actually lethal tech shells being something no one bothers with and won't get in time.
Mechs ain't something you'll get out in time for nukies, and if you do, they'll have dogshit stock parts.
EMP as far as I'm aware doesn't affect nukies at all, they get the advanced EMP shield module.
Mobbing someone with a far better weapon, especially the shotgun or the L6, is not a good idea. same as approaching a de-sword nukie with shield module with a baton wasn't and shouldn't be.

ZephyrTFA commented:
I'm willing to see how this plays out

jlsnow301 commented:
positive reviews + 15 rounds on TM and nothing seemingly mangled
